### PR TITLE
Feature/nil checks

### PR DESCRIPTION
**Pull request by takeoutweight**

### CODE REVIEW

**Comment by dball** [comment]
What fails that requires the second commit? I can't manage to break anything in my tests, except for writing an empty seq of rows which fails by the `assert`.

File: src/google_apps_clj/google_sheets_v4.clj [hunk 1]
```diff
@@ -240,8 +241,9 @@
                                     (.setGridProperties
                                      (-> (GridProperties.)
                                          (.setRowCount (int (count rows)))
-                                         (.setColumnCount (int num-cols)))))))))
-                      (-> (Request.)
+                                          (.setColumnCount (int num-cols)))))))))]
+                      (when (< 0 (count rows))
```
Comment (dball) [inline src/google_apps_clj/google_sheets_v4.clj:245]: It's funny, I don't really mind the prefix nature of these operators, but I also would absolutely have written these two guards as:

`(when-not (seq rows))` and `(when-not (seq (rest rows)))`

Comment (calvis) [reply src/google_apps_clj/google_sheets_v4.clj:245]: y not `empty?`?  is that function unpredictable / worse than `seq` for some reason?

Comment (dball) [reply src/google_apps_clj/google_sheets_v4.clj:245]: Not afaict, it's just purportedly idiomatic clojure; certainly habitual on my part by now.

FWIW `empty?` and `not-empty` are in fact implemented using `seq`.

**Pull request merged**